File: examples/developer-hub-go/get_feed_id.go
package main

import (
	"encoding/hex"
	"fmt"
	"strings"
)

func getFeedID(category, feedName string) string {
	hexFeedName := hex.EncodeToString([]byte(feedName))
	paddedHexString := category + hexFeedName
	if len(paddedHexString) < 42 {
		paddedHexString = paddedHexString + strings.Repeat("0", 42-len(paddedHexString))
	}
	feedId := "0x" + paddedHexString
	fmt.Println(feedId)
	return feedId
}

File: examples/developer-hub-python/get_feed_id.py
def get_feed_id(category: str, feed_name: str) -> str:
    hex_feed_name = feed_name.encode("utf-8").hex()
    padded_hex_string = (category + hex_feed_name).ljust(42, "0")
    return f"0x{padded_hex_string}"


feed_id = get_feed_id("01", "FLR/USD")
print(feed_id)
